test/ipc_os_prov_consumer.c

```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#include <arpa/inet.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include <umf/providers/provider_os_memory.h>
+
+#define INET_ADDR "127.0.0.1"
+#define MSG_SIZE 256
+#define RECV_BUFF_SIZE 1024
+
+// consumer's response message
+#define CONSUMER_MSG                                                           \
+    "This is the consumer. I just wrote a new number directly into your "      \
+    "shared memory!"
+
+/*
+Generally communication between the producer and the consumer looks like:
+- Consumer starts
+- Consumer creates a socket
+- Consumer listens for incoming connections
+- Producer starts
+- Producer's shared memory contains a number: 140722582213392
+- Producer gets the IPC handle
+- Producer creates a socket
+- Producer connects to the consumer
+- Consumer connects at IP 127.0.0.1 and port 47770 to the producer
+- Producer sents the IPC handle to the consumer (24 bytes)
+- Consumer receives the IPC handle from the producer (24 bytes)
+- Consumer opens the IPC handle received from the producer
+- Consumer reads the number from the producer's shared memory: 140722582213392
+- Consumer writes a new number directly to the producer's shared memory: 70361291106696
+- Consumer sents a response message to the producer
+- Consumer closes the IPC handle received from the producer
+- Producer receives the response from the consumer: "This is the consumer. I just wrote a new number directly into your shared memory!"
+- Producer verifies the consumer wrote the correct value (the old one / 2) to the producer's shared memory: 70361291106696
+- Producer puts the IPC handle
+- Consumer shuts down
+- Producer shuts down
+*/
+
+int main(int argc, char *argv[]) {
+    struct sockaddr_in consumer_addr;
```

Comment:
nit: could you put the code responsible for sending/receving the data over the sockets into separate functions? it's a bit hard to review it as is.

Reply:
Done